Split ieeg_labelElectrodesDestrieux.m

hharveygit opened the pull request:
Created ieeg_getLabelXyzDestrieux.m from ieeg_labelElectrodesDestrieux.m
- returns Destrieux labels for input XYZ coordinates

Modified ieeg_labelEletrodesDestrieux.m to call ieeg_getLabelXyzDestrieux.m

^ Both files tested to match outputs from original Destrieux labels.

Also added helper function ieeg_getPairXyzs.m
- returns interpolated XYZ coordinates at the midpoint of electrode pairs

hharveygit committed:
Added functions to return Destrieux label from input XYZ coordinate and to return interpolated XYZ coordinates between pairs of electrodes

hharveygit committed:
Modified ieeg_labelElectrodesDestrieux.m to get labels from ieeg_getLabelXyzDestrieux.m; small edit to nargin condition of ieeg_getLabelXyzDestrieux.m

dorahermes commented on functions/ieeg_labelElectrodesDestrieux.m:
```diff
@@ -1,4 +1,4 @@
-function [t_new] = ieeg_labelElectrodesDestrieux(FSdir,electrodes_tsv_name,saveNew,circleradius)
+function [t_new] = ieeg_labelElectrodesDestrieux(FSdir, electrodes_tsv_name, saveNew, circleradius, hemi)
```
what happens if there are bilateral electrodes?

dorahermes reviewed (commented):
Only question I have is how this deals with bilateral implants, would we run 2x, once for left, once for right sided electrodes?

hharveygit commented:
I haven't thought about any bilateral cases yet. Good point. All of the code I've written so far has been under the assumption of unilaterality. I'll have to modify the code a bit to generalize it to a bilateral situation. The original label Destrieux code actually had left hemisphere hard-coded in it.

Best
-Harvey

Sent from my iPhone

On Nov 13, 2020, at 6:10 PM, dorahermes <notifications@github.com> wrote:

﻿

@dorahermes commented on this pull request.

Only question I have is how this deals with bilateral implants, would we run 2x, once for left, once for right sided electrodes?

________________________________

In functions/ieeg_labelElectrodesDestrieux.m<https://github.com/MultimodalNeuroimagingLab/mnl_ieegBasics/pull/5#discussion_r523294878>:

> @@ -1,4 +1,4 @@
-function [t_new] = ieeg_labelElectrodesDestrieux(FSdir,electrodes_tsv_name,saveNew,circleradius)
+function [t_new] = ieeg_labelElectrodesDestrieux(FSdir, electrodes_tsv_name, saveNew, circleradius, hemi)


what happens if there are bilateral electrodes?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/MultimodalNeuroimagingLab/mnl_ieegBasics/pull/5#pullrequestreview-530490148>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ARAYP37OMYLDLDZFUGC6CZTSPXDEBANCNFSM4TVEQUXQ>.

dorahermes commented on functions/ieeg_getLabelXyzDestrieux.m:
```diff
@@ -0,0 +1,201 @@
+%       
+%       xyzs =              nx3 array of n [x, y, z] coordinates
+%       FSdir =             path to FreeSurfer directory for the current subject; e.g. */derivatives/freesurfer/sub-<subject>
+%       hemi (optional) =   string: 'left', 'L', 'l' or 'right', etc. corresponding to hemisphere of subject. 'left' by default
```
hemi is actually only used here to load the Destrieux colortable, which is hemisphere agnostic, but not for any further processing

dorahermes commented on functions/ieeg_getLabelXyzDestrieux.m:
```diff
@@ -0,0 +1,201 @@
+    %% load files
+    niDestrieux = niftiRead(fullfile(FSdir, 'mri', 'aparc.a2009s+aseg.nii.gz')); % labelled nifti
+
+    [~, ~, colortable_Destrieux] = read_annotation(fullfile(FSdir, 'label', sprintf('%sh.aparc.a2009s.annot', lower(hemi(1)))));
```
Only place here where hemi is used

dorahermes commented:
I guess it currently only uses the hemi tag to load the Destrieux colortable, which is actually hemisphere agnostic. The actual code only looks for the most common label close to an electrode, so we may remove this?

hharveygit commented:
Hi Dora,

Oh yes, you are correct. I'll just hardcode the left hemisphere in then.

Best,
-Harvey
________________________________
From: dorahermes <notifications@github.com>
Sent: Wednesday, November 18, 2020 11:49 AM
To: MultimodalNeuroimagingLab/mnl_ieegBasics <mnl_ieegBasics@noreply.github.com>
Cc: Huang, Harvey <Huang.Harvey@mayo.edu>; Author <author@noreply.github.com>
Subject: [EXTERNAL] Re: [MultimodalNeuroimagingLab/mnl_ieegBasics] Split ieeg_labelElectrodesDestrieux.m (#5)


I guess it currently only uses the hemi tag to load the Destrieux colortable, which is actually hemisphere agnostic. The actual code only looks for the most common label close to an electrode, so we may remove this?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/MultimodalNeuroimagingLab/mnl_ieegBasics/pull/5#issuecomment-729847482>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ARAYP322DZIROVMND5MU7CDSQQCLLANCNFSM4TVEQUXQ>.

hharveygit committed:
Removed hemisphere argument. Hardedcoded left-hemisphere as it does not affect the Colortable loaded

hharveygit commented:
Just pushed the changes!

-Harvey
________________________________
From: Huang, Harvey <Huang.Harvey@mayo.edu>
Sent: Wednesday, November 18, 2020 12:30 PM
To: MultimodalNeuroimagingLab/mnl_ieegBasics <reply@reply.github.com>
Subject: Re: [EXTERNAL] Re: [MultimodalNeuroimagingLab/mnl_ieegBasics] Split ieeg_labelElectrodesDestrieux.m (#5)

Hi Dora,

Oh yes, you are correct. I'll just hardcode the left hemisphere in then.

Best,
-Harvey
________________________________
From: dorahermes <notifications@github.com>
Sent: Wednesday, November 18, 2020 11:49 AM
To: MultimodalNeuroimagingLab/mnl_ieegBasics <mnl_ieegBasics@noreply.github.com>
Cc: Huang, Harvey <Huang.Harvey@mayo.edu>; Author <author@noreply.github.com>
Subject: [EXTERNAL] Re: [MultimodalNeuroimagingLab/mnl_ieegBasics] Split ieeg_labelElectrodesDestrieux.m (#5)


I guess it currently only uses the hemi tag to load the Destrieux colortable, which is actually hemisphere agnostic. The actual code only looks for the most common label close to an electrode, so we may remove this?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/MultimodalNeuroimagingLab/mnl_ieegBasics/pull/5#issuecomment-729847482>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ARAYP322DZIROVMND5MU7CDSQQCLLANCNFSM4TVEQUXQ>.

dorahermes commented:
great thanks!

dorahermes committed:
Merge branch 'master' into master

hharveygit committed:
created ieeg_labelSeizure code

hharveygit committed:
Changes to be committed:
	new file:   functions/expandHyphens.m
		helper function to ieeg_labelSeizure.m
	new file:   functions/ieeg_butterhighpass.m
		to complement ieeg_butterlowpass.m
	modified:   functions/ieeg_labelSeizure.m
		- Incorporated UI, dialogue box edits from Tal
		- Writes/overwrites the seizure_zone column
		- Determines if seizure_zone has been written into by checking if it is numeric (default = all NaN column, which is numeric). Becomes a text column when anything is written into it (including "n/a")

hharveygit committed:
Merge branch 'master' of https://github.com/hharveygit/mnl_ieegBasics

Merge is primarily to update local repo after adding seizure_zone labelling code

dorahermes commented:
thank you, this is great!
